verifier/verifier.go

```diff
@@ -0,0 +1,265 @@
+// Package Verifier provides an implementation of notation.Verifier interface
+package verifier
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/notaryproject/notation-go/dir"
+	"github.com/notaryproject/notation-go/internal/envelope"
+	"github.com/notaryproject/notation-go/notation"
+	"github.com/notaryproject/notation-go/plugin"
+	"github.com/notaryproject/notation-go/plugin/manager"
+	"github.com/notaryproject/notation-go/verifier/trustpolicy"
+)
+
+// verifier implements notation.Verifier
+type verifier struct {
+	TrustPolicyDoc *trustpolicy.Document
+	PluginManager  pluginManager
+}
+
+// pluginManager is for mocking in unit tests
+type pluginManager interface {
+	Get(ctx context.Context, name string) (*manager.Plugin, error)
+	Runner(name string) (plugin.Runner, error)
+}
+
+// New a verifier based on local file system
+func New() (notation.Verifier, error) {
+	// load trust policy
+	policyDocument, err := loadPolicyDocument()
+	if err != nil {
+		return nil, err
+	}
+	if err = policyDocument.Validate(); err != nil {
+		return nil, err
+	}
+
+	return &verifier{
+		TrustPolicyDoc: policyDocument,
+		PluginManager:  manager.New(dir.PluginFS()),
+	}, nil
+}
+
+// NewVerifier creates a new verifier given trustPolicy and pluginManager
+func NewVerifier(trustPolicy *trustpolicy.Document, pluginManager *manager.Manager) (notation.Verifier, error) {
+	if err := trustPolicy.Validate(); err != nil {
+		return nil, err
+	}
+	return &verifier{
+		TrustPolicyDoc: trustPolicy,
+		PluginManager:  pluginManager,
+	}, nil
+}
+
+// TrustPolicyDocument gets the validated trust policy document.
+func (v *verifier) TrustPolicyDocument() (*trustpolicy.Document, error) {
+	if err := v.TrustPolicyDoc.Validate(); err != nil {
+		return nil, err
+	}
```

Comment:
`Validate()` is being called when `New` or `NewVerifier`. Do we still need to call `Validate()` here?